### PR TITLE
Oasis anonymous call

### PR DESCRIPTION
**Pull request by manfer**
I had not the chance to test it yet properly but It would be great if you can have a look to this implementation of https://github.com/irontec/ivozprovider/issues/13 using a prefix. Just to know if it is fine to do it this way or if it is a bad idea.

### CODE REVIEW

**Comment by Kaian** [comment]
Hi @manfer 

Thanks for the pull request!  We will valorate and discuss this internally, because right now we are up-porting all oasis features to artemis and we are not sure how this will be ported.

In artemis, all application servers (asterisk, agis, ...) only handle numbers in E.164 with plus sign, so old outboundprefix logics has been moved to kamailio@users using numeric transformations, that are in charge of transforming what user dialed to that standard format, but are not ready for optional prefixes. The only optional prefix users can choose is ruled by _Routing Tags_ feature (#489) that could be used to choose the desired outgoingDDI used (or even none of them for an anonymous call). Those tags are not implemented for vPBX so porting this feature involves quite some analysis and development.

To sum up, this seems totally ok to me, but will hold this a bit until we have a valid artemis implementation.

Thanks a lot!

**Comment by cruzccl** [comment]
Hi @manfer,

Thank you for taking time to contribute a PR with a new feature, it is highly appreciated.

We have thought many times about developing a feature like this (in fact #13 is one of our oldest issues), maybe not prefix-enabled but something that allows making anonymous calls.

The reason why we haven't done it yet is that we have seen that different carriers provide anonymous calls in a different ways:

- Some need setting anonymous in user-part of From (or PAI) header as your PR does.

- Some others need an additional anonymous-like domain in such headers (e.g. anonymous.invalid).

- Some others need additional Privacy header.

- Etc.

We will eventually implement this in Artemis (as you know we are avoiding scheme changes in Oasis to avoid complicating migration to Artemis), but we need to read thoroughly related RFCs ([RFC 3323](https://tools.ietf.org/html/rfc3323), [RFC 3325](https://tools.ietf.org/html/rfc3325)) before to learn how to implement it correctly, so that it works with all carriers that follow given specifications.

That said, I'm afraid we cannot merge this PR in IvozProvider until we have done all this analysis. As always, you can talk to our commercial department to prioritize this feature in Artemis release.

Regards,

**Comment by manfer** [comment]
Thanks for detail description of the problem.

Besides, I think my implementation lacks the ability to decide which route to use to make the call. If I'm not wrong, it is kamailio the one that decides which provider to use based on the configured outgoing routes. I was thinking of using the prefix on an outgoing route but I think kamailio will get the outgoing call number without the prefix so that outgoing routing including the prefix would be ignored.

**Comment by cruzccl** [comment]
Yes, KamTrunks decides the outgoing carrier for each call and, as you say, the anonymous prefix could collide with numeric transformations and routing logic.

We close this issue then to focus in active ones.

Thank you and regards,
